src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.c

```diff
@@ -788,10 +791,9 @@ static void rr_handover_locked(grpc_exec_ctx *exec_ctx,
                                glb_lb_policy *glb_policy) {
   GPR_ASSERT(glb_policy->serverlist != NULL &&
              glb_policy->serverlist->num_servers > 0);
-
   if (glb_policy->shutting_down) return;
-
   grpc_lb_policy_args *args = lb_policy_args_create(exec_ctx, glb_policy);
+  if (args == NULL) return;
```

Comment:
Simply returning here seems wrong.  Consider the case where we first get an update containing at least one non-drop address, and we then get an update containing only drop addresses.  In this case, I think we should update the RR policy with an empty list, so that it does not continue to maintain a connection to the original non-drop address.

We will still drop all calls regardless, because the code in `pick_from_internal_rr_locked()` checks whether the serverlist entries are drop entries before asking the RR policy to do the pick.  But it seems cleaner to update the RR policy appropriately, so that we don't maintain a subchannel that we don't need.

I think a better fix would be to change `process_serverlist_locked()` to return an empty list of addresses instead of returning NULL.  We can then use that empty list just as we would use any other list, without any special checks for NULL.